### ISSUE TITLE
`useMediaQuery` is not working as expected

### ISSUE THREAD

**Issue by jd-solanki** [vueuse/vueuse#1591]
### Describe the bug

I am using [`usePreferredColorScheme`](https://vueuse.org/core/usepreferredcolorscheme/) composable but I wasn't returning the value I wanted so I dig down and tried to use [`useMediaQuery`](https://vueuse.org/core/usemediaquery/).

I noticed that `useMediaQuery('(prefers-color-scheme: dark)')` is returning `false`. However, if I run below code, I get `DARK MODE...` logged to console.

```ts
if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches)
  console.log('DARK MODE...')
```

**Full code**

```ts
const colorScheme = usePreferredColorScheme()
console.log('colorScheme.value :>> ', colorScheme.value)

const isPreferredDark = useMediaQuery('(prefers-color-scheme: dark)')
console.log('isPreferredDark :>> ', isPreferredDark.value)

if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches)
  console.log('DARK MODE...')
```

**Output**
<img width="636" alt="Screenshot 2022-05-12 at 12 39 09 PM" src="https://user-images.githubusercontent.com/47495003/168012493-3f93f093-53cb-4278-95e8-75ea38b8fbaa.png">

### Reproduction

https://stackblitz.com/edit/vitejs-vite-dlmmn4?file=src/App.vue

### System Info

```Shell
System:
    OS: Linux 5.15 Ubuntu 22.04 LTS 22.04 LTS (Jammy Jellyfish)
    CPU: (6) x64 Intel(R) Core(TM) i5-9400F CPU @ 2.90GHz
    Memory: 1.07 GB / 7.71 GB
    Container: Yes
    Shell: 0.12.1 - /home/jd/.xonsh/.venv/bin/xonsh
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v16.14.2/bin/yarn
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
  Browsers:
    Chrome: 101.0.4951.54
  npmPackages:
    @vueuse/core: ^8.2.6 => 8.4.2 
    vue: ^3.2.33 => 3.2.33
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by alokVishu** [comment]
Yes, I am also feeling the same issue.
+1

**Comment by jd-solanki** [comment]
Keep it open

**Comment by GODLiangCY** [comment]
It should be solved in #2178 . How about trying again with the newest version of VueUse?

**Comment by jd-solanki** [comment]
It is working as expected after trying posted code with the latest version.
